From e57fdd377b3d2be319da7c08675fb8782635ba01 Mon Sep 17 00:00:00 2001
From: Puneet Mishra <puneetm@codeaurora.org>
Date: Tue, 24 Nov 2015 14:45:45 +0000
Subject: [PATCH] NFC: Adding new vendor specific interface to NFC Service

A new interface was added to INfcAdapter for vendor-specific
extensions and additional features. It needs to be implemented
in all implementations of the interface.

Change-Id: I9aa8b94cfbb04133d67c2272c82409cf7db45fec
---
 src/com/android/nfc/NfcService.java | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/com/android/nfc/NfcService.java b/src/com/android/nfc/NfcService.java
index 6ab45a58..c325a8b2 100644
--- a/src/com/android/nfc/NfcService.java
+++ b/src/com/android/nfc/NfcService.java
@@ -1,4 +1,7 @@
 /*
+ * Copyright (c) 2018, The Linux Foundation. All rights reserved.
+ * Not a Contribution.
+ *
  * Copyright (C) 2010 The Android Open Source Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
@@ -1154,6 +1157,13 @@ public class NfcService implements DeviceHostListener {
     }
 
     final class NfcAdapterService extends INfcAdapter.Stub {
+        /**
+         * An interface for vendor specific extensions
+         */
+        public IBinder getNfcAdapterVendorInterface(String vendor) {
+            return null;
+        }
+
         @Override
         public boolean enable() throws RemoteException {
             NfcPermissions.enforceAdminPermissions(mContext);
-- 
2.33.0.windows.2

